File: server/models/wishlist.js
const mongoose = require('mongoose');

const wishlistSchema = new mongoose.Schema({
  userId: { type: String, required: true },
  job_id: { type: String, required: true },
  employer_logo: { type: String, required: true },
  job_title : { type: String, required: true },
  job_employment_type : { type: String, required: true },
  data: [{ type: Object, required: true }],
});

const Wishlist = mongoose.model('Wishlist', wishlistSchema);

module.exports = Wishlist;

File: server/models/wishlist.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Wishlist struct {
	ID                primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	UserID            primitive.ObjectID `bson:"user_id" json:"user_id" validate:"required"`
	JobID             string             `bson:"job_id" json:"job_id" validate:"required"`
	EmployerLogo      string             `bson:"employer_logo" json:"employer_logo" validate:"required"`
	JobTitle          string             `bson:"job_title" json:"job_title" validate:"required"`
	JobEmploymentType string             `bson:"job_employment_type" json:"job_employment_type" validate:"required"`
	Data              interface{}        `bson:"data" json:"data" validate:"required"` // Stores any type of object (like `Object` in Mongoose)
}
